### ISSUE TITLE
Empty lines when changing indentation

### ISSUE THREAD

**Issue by elliterate** [rubocop-hq/ruby-style-guide#436]
I see (and abide by) a tendency to avoid empty lines at the beginning and end of a particular indentation level, e.g., within a class or method definition:

``` ruby
# bad
class Foo

  def bar

    puts "bar"

  end

end

# good
class Foo
  def bar
    puts "bar"
  end
end
```

I didn't see a style guide entry for this. Is this a reasonable thing to add?

**Comment by onebree** [comment]
TLDR: I think this is personal preference.

If `bar` was 10-15 LOC, I would see it as not as readable without using white space as a border. I do not add an empty line after a `def` line, but I like to see each piece of the code in blocks. I am sure others do -- it is more bite-sized to understand and quickly glance at. I would say that after calling `class` or `module`, skipping a line is optional, because such can be divided into smaller pieces. If you have a method definition, that cannot be divided smaller (unless it has several actions/steps, like validate/save/commit). But I do not see this is a style that, if flip-flopped throughout an app, would make a noticeable difference to anyone.

**Comment by elliterate** [comment]
It looks like rubocop already has style checks for these:

```yml
Style/EmptyLineBetweenDefs:
  Description: Use empty lines between defs.
  StyleGuide: https://github.com/bbatsov/ruby-style-guide#empty-lines-between-methods
  Enabled: true
  AllowAdjacentOneLineDefs: false
Style/EmptyLinesAroundBlockBody:
  Description: Keeps track of empty lines around block bodies.
  Enabled: true
  EnforcedStyle: no_empty_lines
  SupportedStyles:
  - empty_lines
  - no_empty_lines
Style/EmptyLinesAroundClassBody:
  Description: Keeps track of empty lines around class bodies.
  Enabled: true
  EnforcedStyle: no_empty_lines
  SupportedStyles:
  - empty_lines
  - no_empty_lines
Style/EmptyLinesAroundModuleBody:
  Description: Keeps track of empty lines around module bodies.
  Enabled: true
  EnforcedStyle: no_empty_lines
  SupportedStyles:
  - empty_lines
  - no_empty_lines
Style/EmptyLines:
  Description: Don't use several empty lines in a row.
  Enabled: true
Style/EmptyLinesAroundAccessModifier:
  Description: Keep blank lines around access modifiers.
  Enabled: true
Style/EmptyLinesAroundMethodBody:
  Description: Keeps track of empty lines around method bodies.
  Enabled: true
```

**Comment by elliterate** [comment]
On a personal note, I've worked in a code base where there was a consistent pattern of adding empty lines around class, module, method, and block bodies, and I felt it impaired the readability of the code. It effectively doubled the length of the files—especially as method definitions were kept small—giving you much less information at a glance than files that consistently excluded such empty lines.

And while I could see how it might help set apart larger method definitions, I would think those would be more likely to fail the Assignment Branch Condition size checks.

**Comment by bbatsov** [comment]
> I didn't see a style guide entry for this. Is this a reasonable thing to add?

Thought we had a rule for this. Guess this is not the case, so go ahead and add one to the code layout section.

**Comment by pirj** [comment]
Covered in https://github.com/rubocop-hq/ruby-style-guide#empty-lines-around-bodies
